spatialprofilingtoolbox/ondemand/providers/pending_provider.py

```diff
@@ -0,0 +1,194 @@
+"""Abstract class for providers that need to wait for on demand calculations to complete."""
```

Comment:
I am really glad you created this abstract class. It is very simplifying, and I think it would be even if we only had 1 concrete implementation.

Reply:
I'm considering converting all those static methods to object methods and moving all the `feature_specification` arguments into object state as a member variable...